gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/MultiHopsFlowToJobSpecCompiler.java

```diff
@@ -309,9 +309,10 @@ private JobSpec buildJobSpec (ServiceNode sourceNode, ServiceNode targetNode, UR
     }
 
     // Add flow execution id for this compilation
-    long flowExecutionId = System.currentTimeMillis();
-    jobSpec.setConfig(jobSpec.getConfig().withValue(ConfigurationKeys.FLOW_EXECUTION_ID_KEY,
-        ConfigValueFactory.fromAnyRef(flowExecutionId)));
+    if (!jobSpec.getConfig().hasPath(ConfigurationKeys.FLOW_EXECUTION_ID_KEY)) {
```

Comment:
Can you add this check to AbstractPathFinder#findPath() as well? There you would check if the FlowSpec already has the flowExecutionId.